js/ui/panel.js

```diff
@@ -951,6 +952,14 @@ const Panel = new Lang.Class({
         this._toggleMenu(this.statusArea.dateMenu);
     },
 
+    closeActiveMenu: function() {
```

Comment:
Maybe, but in that case we'd have to access `this._panel.menuManager` from there since we want to close any popup that is open anywhere in the panel (including the system indicator), not just popup menus from the AppIconBar itself.